src/main/java/net/pms/network/HTTPServer.java

```diff
@@ -40,35 +43,97 @@
 
-	public NetworkInterface getNetworkInterface() {
-		return networkInterface;
+	/**
+	 * @return The socketAddress used by this server instance
+	 */
+	public InetSocketAddress getSocketAddress() {
+		return socketAddress;
 	}
 
-	// use getNetworkInterface()
-	@Deprecated
-	public NetworkInterface getNi() {
-		return getNetworkInterface();
-	}
+	public HTTPServer() {
+		int tmpPort = configuration.getServerPort();
+		if (tmpPort < 1 || tmpPort > 65535) {
+			LOGGER.warn("Configured HTTP port outside of valid range (1-65535), using default port ({}) instead", PmsConfiguration.DEFAULT_SERVER_PORT);
+			port = PmsConfiguration.DEFAULT_SERVER_PORT;
+		} else {
+			port = tmpPort;
+		}
+
+		String tmpHostname = configuration.getServerHostname();
+		InetSocketAddress tmpSocketAddress = null;
+
+		if (StringUtils.isNotBlank(tmpHostname)) {
+			LOGGER.info("Using forced address \"{}\"", tmpHostname);
+			InetAddress inetAddress = null;
+			try {
+				inetAddress = InetAddress.getByName(tmpHostname);
+			} catch (UnknownHostException e) {
+				tmpHostname = null;
+				LOGGER.error("Hostname {} cannot be resolved, ignoring parameter: {}", tmpHostname, e.getMessage());
+				LOGGER.trace("", e);
+			}
+
+			if (inetAddress != null) {
+				LOGGER.info("Using address {} resolved from \"{}\"", inetAddress.getHostAddress(), tmpHostname);
+				tmpSocketAddress = new InetSocketAddress(inetAddress, port);
+				networkInterfaces.addAll(NetworkConfiguration.getInstance().getNetworkInterfaces(inetAddress));
+			}
+		}
 
-	public HTTPServer(int port) {
-		this.port = port;
+		if (tmpSocketAddress == null) {
+			InterfaceAssociation interfaceAssociation = null;
+			if (StringUtils.isNotEmpty(configuration.getNetworkInterface())) {
+				interfaceAssociation = NetworkConfiguration.getInstance().getAddressForNetworkInterfaceName(configuration.getNetworkInterface());
+			}
+
+			if (interfaceAssociation != null) {
+				InetAddress inetAddress = interfaceAssociation.getAddr();
+				networkInterfaces.add(interfaceAssociation.getIface());
+				LOGGER.info(
+					"Using address {} found on network interface: {}",
+					inetAddress,
+					interfaceAssociation.getIface().toString().trim().replace('\n', ' ')
+				);
+				socketAddress = new InetSocketAddress(inetAddress, port);
+			} else {
+				networkInterfaces.addAll(NetworkConfiguration.getInstance().getRelevantNetworkInterfaces());
+				LOGGER.info("Using all addresses");
+				socketAddress = new InetSocketAddress(port);
+			}
+		} else {
+			socketAddress = tmpSocketAddress;
+		}
+
+		if (StringUtils.isBlank(tmpHostname) && socketAddress != null) {
+			if (socketAddress.getAddress().isAnyLocalAddress()) {
+				try {
+					tmpHostname = InetAddress.getLocalHost().getHostName();
+				} catch (UnknownHostException e) {
+					tmpHostname = InetAddress.getLoopbackAddress().getHostName();
+				}
+			} else {
+				tmpHostname = socketAddress.getHostString();
+			}
+		}
+		hostname = tmpHostname;
 	}
 
 	public String getURL() {
```

Comment:
@Nadahar that problem was related to the hostname using and it was only the description how it  has an influence to the UMS renderers recognition. Now you fixed it and the problem is gone.